polyfill/intersectionobserver-polyfill.js

```diff
@@ -0,0 +1,250 @@
+/*
+Copyright 2016 Google Inc. All Rights Reserved.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+   http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+// TODO: rootMargin are completely ignored for now
+(function(scope) {
+  if('IntersectionObserver' in window) {
+    return;
+  }
```

Comment:
Polyfill service prefers not to include feature tests inside the polyfill, because you may want to apply a polyfill over a native implementation in some cases, and detects are often too generous (this is actually a perfect case in point, because it will pass on the existing partial implementation in Chrome - though I realise it's behind a flag so not - yet - an issue in the wild)